### ISSUE TITLE
Android box a95x (android 6 s905x) original remote not working in SPMC

### ISSUE THREAD

**Issue by dude-nl** [koying/SPMC#437]
Hi all,

I have a lot of experience with Android boxes but can't figure this one out.
In Kodi 16.1 and Android the remote works fine but in SPMC 16.4.2 the remote is not working. Only is I press the air moude button, the airmouse works, but not the normal mode (cursor etc).

Can somebody help me with this?

**Comment by dude-nl** [comment]
Just plugged in regular keyboard and mouse. Keyboard functions not working in homescreen. Switched to Confluence skin and it's working. Issue is in homescreen (menu's are working) of Re-Thouched screen.

**Comment by koying** [comment]
Why Re-touched?
As the name implies, that's for touchscreens...

**Comment by dude-nl** [comment]
Because it's the standard skin when you start SPMC and you can not change it if the remote is not working...

**Comment by koying** [comment]
> Because it's the standard skin when you start SPMC

It only is if the box firmware is stupid enough to announce itself as having a touchscreen ;)
You can't use it on a TV screen+remote

**Comment by RobBrownNZ** [comment]
> It only is if the box firmware is stupid enough to announce itself as having a touchscreen ;)

Aha, that answers my question as well... yes, the A95X boxes (both S905 and S905X-based) obviously are that stupid :)

**Comment by koying** [comment]
Yeah. Those firmwares are just modified tablet ones, and the "engineers" in charge cannot be bothered to do things properly. Actually, from experience, they barely have a clue about android at all, being mostly kernel and/or linux devs ;)

**Comment by friedretinas** [comment]
@koying, thank you for all of your hard work. 

I assumed that I could create a work-around for this issue by removing skin.re-touched since the box is shipped w/ root access, and ships with the re-touched skin removed from the KODI installation, When I observed that KODI never tries to reinstall the skin, even after a (Android) Settings -> KODI -> Clear Data is performed before re-loading KODI, I figured that the same would apply to SPMC.  So I installed Total Commander (root file manager) and then:
1. navigated to  **_/data/data/com.semperpax.spmc16/cache/apk/assets/addons_**
2. deleted  **_skin.re-touched_**
3. verified the deletion
4. (Android) Settings -> SPMC -> Clear Data

But the next time I loaded up SPMC, it reinstalled skin.re-touched and then selected it as the default skin. I'm **not** complaining that it does this. I'm just pointing out that on this same box that KODI does **not** reinstall the Re-Touched skin and therefore it starts up w/ Confluence after a "Clear Data" has been performed. 

For the most part I'm just posting this to let A95X owners (and possibly other boxes w/ poor firmware) know that if the a "Clear Data" is executed on SPMC, the next time it's launched, the default skin will revert to Re-Touched -- and that even taking the time to delete skin.re-touched w/ a file manager does not provide a work-around. If I'm wrong about this, then please correct me.

So for those of you who will test out multiple builds on this box and have become accustomed to using "Clear Data" (since it's the quickest, simplest way) to reset everything anew, you're probably going to want to use KODI **while** your testing those builds out since it will not attempt to install the Re-Touched skin and will always default to Confluence after a data wipe.

For everything else other than the aforementioned, I prefer SPMC over KODI.  I agree w/ @koying that the firmware is on most of these Chinese made TV Boxes (like the A95X) appears to be written by "engineers" who either lack diligence or there just isn't enough of them working at companies like NEXBOX to handle the workload. The firmware is the most likely the real culprit here. Since NEXBOX does have a reasonably decent forum at: http://www.inexbox.com/forum/  maybe if enough people weigh in on their forum, they will take the time to correct their firmware.

**Comment by friedretinas** [comment]
UPDATE:  I fully unsintalled the (reduced size) version of KODI that NEXBOX pre-installs on the A95X and then installed the latest KODI version using Google Playstore. Even though Re-Touched skin was included w/ this full KODI install, KODI **still** defaults to Confluence.

**So even though both SPMC & KODI are being run on the same A95X, apparently which skin should be the default apparently is simply a matter of a design philosphy difference between Koying and the KODI Team.**
